src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -204,6 +216,13 @@ class CoreWorkerDirectTaskSubmitter {
   absl::flat_hash_map<SchedulingKey, std::deque<TaskSpecification>> task_queues_
       GUARDED_BY(mu_);
 
+  // For each SchedulingKey, a pair of unsigned integers keeps track of
+  // (1) how many worker leases have been granted to execute tasks with
+  //     the current SchedulingKey
+  // (2) how many tasks are in flight to all the workers from (1)
+  absl::flat_hash_map<SchedulingKey, std::pair<uint32_t, uint32_t>> worker_info_
```

Comment:
Discussed offline about refactoring this class to squash all of the fields that are currently related to `SchedulingKey` into one hashmap. The value should be a new struct that includes:
- these new counts
- the workers that are currently leased to us with that scheduling key (`worker_to_lease_entry_`)
- whether there is a pending lease request for that key (`pending_lease_requests_`)
- the tasks queued with that key (`task_queues_`)